core/src/main/java/org/apache/accumulo/core/security/crypto/EncryptionStrategy.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.accumulo.core.security.crypto;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+
+public interface EncryptionStrategy {
+
+  /**
+   * Where in Accumulo the on-disk file encryption takes place.
+   */
+  enum Scope {
+    WAL, RFILE;
+  }
+
+  /**
+   * Initialize the EncryptionStrategy.
+   *
+   * @param encryptionScope
+   *          where the encryption takes places
+   * @return true if initialization was successful
+   * @since 2.0
+   */
+  boolean init(Scope encryptionScope, AccumuloConfiguration configuration) throws Exception;
```

Comment:
These should throw a more narrow exception type.

Reply:
Did you have something in mind?  I wanted to make the interface flexible enough to handle whatever work needs to be done during init... which could be a lot of different things.

Reply:
:man_shrugging: